pkg/services/app/app.go

```diff
@@ -0,0 +1,31 @@
+package app
+
+import (
+	"github.com/weaveworks/weave-gitops/pkg/flux"
+	"github.com/weaveworks/weave-gitops/pkg/git"
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+)
+
+type AppService interface {
+	Add(params AddParams) error
+}
+
+type App struct {
+	git          git.Git
+	flux         flux.Flux
+	kube         kube.Kube
+	gitProviders gitproviders.GitProviderHandler
+}
+
+func New(git git.Git, flux flux.Flux, kube kube.Kube, gitProviders gitproviders.GitProviderHandler) *App {
+	return &App{
+		git:          git,
+		flux:         flux,
+		kube:         kube,
+		gitProviders: gitProviders,
+	}
+}
+
+// Make sure App implements all the required methods.
+var _ AppService = &App{}
```

Comment:
Seems a bit odd. Something else will complain when you try to use the `App` as an `AppService` if it doesn't implement the interface.

Reply:
haha I know, it's a kinda trick the community has, but it will only complain if that is injected or attributed using explicitly somewhere. Like in https://github.com/weaveworks/weave-gitops/pull/336/files#diff-bc17604d5b75b57fd65e69e1996c5c225005be132e891dd100816f4b66a74421R79, unless I set like `var appService AppService = app.New(gitClient, fluxClient, kubeClient, gitProviders)` there won't be complaints about. 

And since using implicit types is the norm, people usually do that to have to guarantee implementation.